storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -577,288 +409,265 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
-                  .setValue(commandLineShim.getCommandLine(details.getId())))
-              .setContainer(
-                  ContainerInfo.newBuilder()
-                      .setType(ContainerInfo.Type.DOCKER)
-                      .setDocker(
-                          ContainerInfo.DockerInfo.newBuilder()
-                              .setImage(_container.get())
-                              .setNetwork(ContainerInfo.DockerInfo.Network.HOST)
-                              .setForcePullImage(true)
-                              .build()
-                      )
-                      .build()
-            );
-        } else {
-          executorInfoBuilder
-              .setCommand(CommandInfo.newBuilder()
-                  .addUris(URI.newBuilder().setValue((String) _conf.get(CONF_EXECUTOR_URI)))
-                  .addUris(URI.newBuilder().setValue(configUri))
-                  .setValue(commandLineShim.getCommandLine(details.getId())));
-        }
+        try {
+          List<ResourceEntry> scalarResourceEntryList = null;
+          List<ResourceEntry> rangeResourceEntryList = null;
+
+          if (subtractExecutorResources) {
+            scalarResourceEntryList = offerResources.reserveAndGet(ResourceType.CPU, new ScalarResourceEntry(executorCpu));
+            executorCpuResources.addAll(createMesosScalarResource(ResourceType.CPU, scalarResourceEntryList));
+            scalarResourceEntryList = offerResources.reserveAndGet(ResourceType.MEM, new ScalarResourceEntry(executorMem));
+            executorMemResources.addAll(createMesosScalarResource(ResourceType.MEM, scalarResourceEntryList));
+          } else {
+            // TODO(ksoundararaj): Do we need to ensure consistent roles for all executors belonging to same topology?
+            // If so, how do we do that?
+            executorCpuResources.add(createMesosScalarResource(ResourceType.CPU, new ScalarResourceEntry(executorCpu)));
+            executorMemResources.add(createMesosScalarResource(ResourceType.MEM, new ScalarResourceEntry(executorMem)));
+          }
 
-        LOG.info("Launching task with Mesos Executor data: < {} >", executorDataStr);
-        TaskInfo task = TaskInfo.newBuilder()
-            .setName(taskName)
-            .setTaskId(taskId)
-            .setSlaveId(offer.getSlaveId())
-            .setExecutor(executorInfoBuilder.build())
-            .addAllResources(workerCpuResources)
-            .addAllResources(workerMemResources)
-            .addAllResources(workerPortsResources)
-            .build();
+          /**
+           *  Autostart logviewer cannot be supported. If we try to fetch a random port P
+           *  from the list of available ports and assign it to the logviewer, its possible that
+           *  one of the workers is already assigned to that port. So I am not supporting auto starting
+           *  logviewer at this point
+           *
+          if (MesosCommon.autoStartLogViewer(mesosStormConf)) {
```

Comment:
kill or prettify please.  I think there are some users that are ok with the brokenness of logviewer, because they only run 1 topology so it's "ok" in its current state.  But it's *really* broken right now...